go/state/state_db_test.go

```diff
@@ -4143,6 +4143,71 @@ func TestSlotIdOrder(t *testing.T) {
 	}
 }
 
+func TestStateDB_SetTransientState_RollbackRemovesValue(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := CreateStateDBUsing(mock)
+
+	s := db.Snapshot()
+	db.SetTransientState(address1, key1, val1)
+
```

Comment:
here you should test that the value is present. If `SetTransientState` was implemented as no-op, this test would still pass.